packages/auth/src/mfa/assertions/totp.ts

```diff
@@ -155,10 +158,21 @@ export class TotpMultiFactorAssertionImpl
 
   /** @internal */
   _finalizeSignIn(
-    _auth: AuthInternal,
-    _mfaPendingCredential: string
+    auth: AuthInternal,
+    mfaPendingCredential: string
   ): Promise<FinalizeMfaResponse> {
-    throw new Error('method not implemented');
+    _assert(
+      typeof this.enrollmentId !== 'undefined' &&
+        typeof this.otp !== 'undefined',
```

Comment:
No need to check `this.otp`. It's not marked as an optional param and since we're on TypeScript we can rely on the type checker to ensure this

Reply:
Fixed